LibraryApp/src/components/show-book.js

```diff
@@ -1,28 +1,30 @@
 import React from 'react';
-import {View, Text, Pressable} from 'react-native';
+import {connect} from 'react-redux';
+import {ScrollView, View, Text, Pressable} from 'react-native';
 import {Colors, Global, Spacing, Typography} from 'stylesheets';
+import {calculateDistance} from 'helpers/application';
 import {libraryForm} from 'constants/navigators';
 import {CustomImage} from 'components/shared/common';
 
-const {
-  flex1,
-  horizontalCenter,
-  borderWidth1,
-} = Global;
+const {flex1, horizontalCenter} = Global;
 const {gochiFont, comicFont, fs18, fs20, fs25, bold} = Typography;
 const {py20, p20, mb5, mb20, mtAuto} = Spacing;
-const {grey, darkBlue, lightGreen, white, maroon, darkGrey} = Colors;
+const {grey, darkBlue, lightGreen, white, maroon} = Colors;
 
-const ShowBook = ({route, navigation}) => {
-  const {book} = route.params || {};
-  const {name, author, publisher, price, image} = book;
+const defaultObj = {};
+
+const ShowBook = ({route, navigation, currentLocation}) => {
+  const {book} = route.params || defaultObj;
+  const {image, name, author, publisher, price, location} = book;
+  const {latitude: endLat, longitude: endLong} = location || defaultObj;
```

Comment:
I think we should take out this logic in another component, as this value will change a lot and we don't need to update the whole component because of this

Reply:
Created separate component for BookDistance.